### DISCUSSION TITLE
Previous efforts

### DISCUSSION THREAD

**Discussion by pi0** [pi0/config-dir#5] [General]
> [!NOTE]
> Here is a collection of inspirations collected from different sources.
>
> See [`.config` dir support tracker](https://github.com/pi0/config-dir/discussions/6) for a list of tools and config loaders that support `.config/` dir or might do.

## How it started

A few months ago, I was inspired by seeing [tweet](https://twitter.com/DanaWoodman/status/1699134345196495182) by @danawoodman about this problem a few months ago and while he was [~~convinced~~ settled](https://twitter.com/DanaWoodman/status/1699535674867949905) to use file nesting, the real problem still exists with actual files not being organized. Finally, when I was dealing with this problem again when building a minimal documentation tool, I decided to push it again!

### Media

- https://twitter.com/iansu/status/1126512410313891841 by @iansu 
- https://twitter.com/DanaWoodman/status/1699134345196495182 by @danawoodman
- https://twitter.com/_pi0_/status/1757383840270532894
- https://twitter.com/_pi0_/status/1757418709574939058
- https://www.youtube.com/watch?v=wYdnJPYFTIE by @t3dotgg

### XDG Spec

[XDG](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) uses `~/.config` as a standard dir (while it allows to be overridden, it may not be feasible for this proposal: [discussion](https://github.com/pi0/config-dir/discussions/4)).

### `dot-config`

_(thanks @webpro for reference)_

[dot-config](https://github.com/dot-config/dot-config.github.io) is a previous effort to push the `.config/` idea by @ssbarnea.

### `proload`

_(thanks @webpro for reference)_

[preload](https://github.com/natemoo-re/proload) by @natemoo-re is another configuration loader that supports `config/` dir. (see [alternative directory names](https://github.com/pi0/config-dir/discussions/3) discussion.) 

There is an [open issue](https://github.com/natemoo-re/proload/issues/34) proposing to also support `.config` directory.

### Discussions in `nodejs/tooling`

_(thanks @webpro for reference)_

- nodejs/tooling#71 (by @vjpr)
- nodejs/tooling#79 (by @boneskull)

### Deno

_(thanks to @t3dotgg for reference)_

@lambtron in an [Article](https://deno.com/blog/node-config-hell) mentioned about all the pain about current configuration files and the value of having best defaults that work out of the box.

**Comment by webpro** [comment]
Nice initiative! More related resources:

- https://dot-config.github.io/
- https://github.com/nodejs/tooling/issues/79
- https://github.com/natemoo-re/proload

**Comment by danawoodman** [comment]
woah i'm famous! 🕺🪩

PS I wasn't "convinced" I settled for it 😂

**Comment by nicolehollant** [comment]
Love this!! I threw together a silly package to let me (sorta) use a config dir a while back, it just lets you create/destroy symlinks for stuff, but it's something ig: https://github.com/nicolehollant/move-stuff 😅
